### DISCUSSION TITLE
Inconsistency in search results between using API and installing on personal server

### DISCUSSION THREAD

**Discussion by xinyuluo** [osm-search/Nominatim#2514] [Q&A]
Hi, I'm new to Nominatim. I tried to install Nominatim on my own server but it returns different results from searching directly on https://nominatim.openstreetmap.org/ui/. Particularly, 

- when queries are about specific location or entity (e.g. white house, raw taipei, etc.) results generated from my server is consistent with using the API directly;
- when searching for generic queries like "food in france", "hospitals in singapore", results are quite different. to give an example,
for the same query as https://nominatim.openstreetmap.org/ui/search.html?q=hospitals+in+singapore, on my own server I get:
```
[{"place_id":91267744,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright","osm_type":"node","osm_id":8763326578,"boundingbox":["17.4365346","17.4366346","78.664295","78.664395"],"lat":"17.4365846","lon":"78.664345","display_name":"Shri Wazra Multi Speciality Hospitals, NH163, Rajeev Gruha Kalpa, Annoji Guda, Ghatkesar mandal, Medchal–Malkajgiri, Telangana, 501301, India","place_rank":30,"category":"amenity","type":"hospital","importance":0.201}]
```
which is a hospital in India.

For the context, I am using version 3.7.2 and imported planet OSM data from Oct 2021. I wonder if it's due to difference with the latest version (4.0.0) (how to check if it's the case?), or I have missed something in my installation. I have also installed Wikipedia rankings according to the installation guide. Thank you so much!

**Comment by lonvia** [comment]
Have you installed the [special phrases](https://nominatim.org/release-docs/3.7.2/admin/Import/#tuning-the-database)? Those are required for the kind of category queries you are doing.

That said, there are bound to be larger differences between a 3.7.2 installation and what is live on nominatim.openstreetmap.org because the way we index search terms has changed significantly.

**Reply by xinyuluo** [reply]
Thanks for the fast reply! 

Yes, I imported those special key phrases after planet data import.

If I understand correctly, what is live on nominatim.openstreetmap.org is based on 4.0.0 (or some more advanced unreleased version?), and query indexing algorithm has changed significantly from 3.7.2 (what im using), hence the discrepancies in search result? 

If this understanding is correct, I'll try to update to 4.0.0 and see if that improves performance.

**Reply by lonvia** [reply]
It might or might not improve it. It is hard to say without looking at the debug output on your machine to check why exactly the query gave the wrong result. If you do a 4.0 import to get the same results, please be aware that our production servers run with the ICU tokenizer which is not yet the default. See [the release blog post](https://nominatim.org/2021/11/03/release-40.html) for details
